remove pyqt6 dependency and re-worked setup cfgs

amitabhverma opened the pull request:
- removed pyqt6 dependency
- streamline different setup needs to address mehta-lab/waveorder#194
- added position metadata to info label mehta-lab/waveorder#191
- bug report template to include relevant logs which are created next to the output dir, name of the dataset, yaml file(s) if encountering reconstruction errors
- CLI can be used without QtBindings (PyQT or PySide) but will show a warning

amitabhverma committed:
remove pyqt6 dependency and re-worked setup cfgs

- removed pyqt6 dependency
- streamline different setup needs to address #498
- added position metadata to info label #500
- bug report template to include relevant logs which are created next to the output dir, name of the dataset, yaml file(s) if encountering reconstruction errors
- CLI can be used without QtBindings (PyQT or PySide) but will show a warning

amitabhverma committed:
Update pytests.yml

changed install flag to "all"

talonchandler committed:
update tox extras

talonchandler committed:
restore dev dependencies for tests

talonchandler commented on recOrder/plugin/__init__.py:
```diff
@@ -1,11 +1,16 @@
-        "Please remove PyQt5 from your environment with `pip uninstall PyQt5`"
-    )
-elif qtpy.PYSIDE2:
-    raise RuntimeError(
-        "Please remove PySide2 from your environment with `pip uninstall PySide2`"
-    )
+    if qtpy.PYQT5:
+        raise RuntimeError(
+            "Please remove PyQt5 from your environment with `pip uninstall PyQt5`"
+        )
+    elif qtpy.PYSIDE2:
+        raise RuntimeError(
+            "Please remove PySide2 from your environment with `pip uninstall PySide2`"
+        )
+
+except RuntimeError as error:
+    if type(error).__name__ == "QtBindingsNotFoundError":
+        print("WARNING: QtBindings (PyQT or PySide) was not found for GUI")
```
Is this still needed?

talonchandler replied:
Sounds good to me. Making this more general makes a lot of sense.

talonchandler replied:
Thanks for simplifying. Looks good.

talonchandler reviewed (changes requested):
I tested three installations and all behaved as I expected:
1. `pip install .`. As expected, the CLI worked, the GUI did not (does not appear in CLI list, no napari available). 
2. `pip install .[all]`. As expected, both the CLI and GUI work. 
3. An existing napari installation. `pip install napari[all]` (installs `PyQt5`) then `pip install .[all]`. As expected, both the CLI and GUI work. 

I left a few comments below. Also, please update the README and `/docs` installation instructions to reflect these changes.

amitabhverma committed:
refined dependencies, updated docs

- moved pycromanager and magicgui to [all] flag

amitabhverma committed:
re-worked essential dependencies and catching MM connection error

- mostly applies when barebone recOrder is installed
- in order to lead a user to install the correct missing package (pycromanager)

amitabhverma committed:
MM connection failed msg

amitabhverma committed:
added Reconstruction Tab section to docs

talonchandler reviewed (approved):
Looks good, thanks @amitabhverma. Just retested the tests I described above, and all's working. Merging.
